Tests/AmplitudeTests/Utilities/IdentifyInterceptorTests.swift

```diff
@@ -235,21 +190,26 @@ final class IdentifyInterceptorTests: XCTestCase {
         XCTAssertNotNil(events[0].userProperties)
         XCTAssertTrue(getDictionary(events[0].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-1", "key-2": "value-2"]]))
 
-        // identify with $add should receive merged user properties
+        // event with $add should not be intercepted
         let e2 = interceptor.intercept(event: testEvent2)
         XCTAssertNotNil(e2)
         XCTAssertEqual(e2!.eventType, "$identify")
         XCTAssertNotNil(e2!.userProperties)
-        XCTAssertTrue(getDictionary(e2!.userProperties!).isEqual(to: ["$set": ["key-1": "value-1-2", "key-2": "value-2", "key-3": "value-3"], "$add": ["add-1": "add-2"]]))
+        XCTAssertTrue(getDictionary(e2!.userProperties!).isEqual(to: ["$set": ["key-1": "value-1-2", "key-3": "value-3"], "$add": ["add-1": "add-2"]]))
+
+        // intercept should be transferred to event storage
         events = identifyStorage.events()
         XCTAssertEqual(events.count, 0)
         events = storage.events()
-        XCTAssertEqual(events.count, 0)
+        XCTAssertEqual(events.count, 1)
+        XCTAssertNotNil(events[0].userProperties)
+        XCTAssertTrue(getDictionary(events[0].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-1", "key-2": "value-2"]]))
     }
 
     func testInterceptIdentifyAndStandardEvent() {
         let testEvent1 = BaseEvent(eventType: "$identify", userProperties: ["$set": ["key-1": "value-1-1", "key-2": "value-2"]])
-        let testEvent2 = BaseEvent(eventType: "someEvent", userProperties: ["key-1": "value-1-2", "key-3": "value-3"])
```

Comment:
I guess by changing the order we also change what we are testing here.

Reply:
I believe the order wasn't changed here. I added second $identify event only to cover more logic. Maybe it makes sense to split the test - to test one $identify event, to test multiple $identify events.